streamcopy/copy.go

```diff
@@ -0,0 +1,57 @@
+package streamcopy
+
+import (
+	"bufio"
+	"io"
+	"time"
+
+	"github.com/containers/image/types"
+)
+
+const megaByte = float64(1024 * 1024)
+const readerSize = 32768
+
+// WithProgress implements io.Copy with a buffered reader, then measures
+// reports to the offsetFunc with the count processed at the interval.
+// If io.EOF is not returned then the error is returned. Otherwise, it is nil.
+// WithProgress closes the writer after finishing.
+func WithProgress(writer io.WriteCloser, reader io.Reader, artifact types.BlobInfo, interval time.Duration, signal chan types.SignalProperties) error {
+	defer writer.Close()
+	rd := bufio.NewReaderSize(reader, readerSize)
```

Comment:
Why is the buffering useful in here? Immediately below we are creating our own buffer.

(I’m not saying it isn’t, I just can’t immediately see that it is.)